src/client/lib/controllers/usersServices.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:convert';
+import 'dart:ffi';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
+
+class usersServices{
+
+  Future<bool> loginUser(String username, String password) async {
+
+    final url = Uri.parse('http://10.0.2.2:8080/users/$username/login');
+
+    final body = jsonEncode({
+      'username': username,
+      'password': password,
+    });
+
+    // Send POST request to /login
+    final response = await http.post(
+      url,
+      headers: {'Content-Type': 'application/JSON'},
+      body: body,
+    );
+
+    print(response.statusCode);
+    print(response.body);
+
+    if (response.statusCode == 200) {
+      final data = jsonDecode(response.body);
+      final sessionId = data['session_id'];
+
+      // Store the session ID in shared preferences
+      final prefs = await SharedPreferences.getInstance();
+      await prefs.setString('sessionId', sessionId);
+
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  Future<bool> logoutUser(String username) async {
+    final prefs = await SharedPreferences.getInstance();
+    final sessionId = prefs.getString('session_id');
+
+    await prefs.remove('session_id');
+    if (sessionId == null) {
+      //No session ID found, user is already logged out
+      return false;
+    }
+
+    final url = Uri.parse('http://10.0.2.2:8080/users/$username/logout');
+
+    // Send POST request to /logout with session ID in headers
+    final response = await http.post(
+      url,
+      headers: {
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer $sessionId',
+      },
+    );
+
+    if (response.statusCode == 200) {
+      // Remove session ID from shared preferences
+      await prefs.remove('session_id');
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  Future<bool> getUserData(String username) async {
```

Comment:
In this case you could make the argument that you want the data of another user so it's fine.